subprojects/core/src/main/java/org/gradle/configuration/project/LifecycleProjectEvaluator.java

```diff
@@ -46,14 +43,13 @@ public void evaluate(final ProjectInternal project, final ProjectStateInternal s
             return;
         }
 
-        String displayName = "project " + project.getIdentityPath().toString();
-        buildOperationExecutor.run(BuildOperationDetails.displayName("Configure " + displayName).name(StringUtils.capitalize(displayName)).build(), new Action<BuildOperationContext>() {
+        domainObjectConfigurator.configure(project, new Action<ProjectInternal>() {
```

Comment:
Another reason not to use a service here is that it creates yet another indirection layer with another `Action` instance, even though we want to run this operation immediately.